drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/main/java/org/drools/workbench/services/verifier/plugin/client/CellUpdateManagerBase.java

```diff
@@ -64,6 +78,20 @@ public CellUpdateManagerBase(final Index index,
      * @return Returns true if the cell content was found and the value changed.
      */
     public boolean update() {
+        boolean isAttribute = false;
+        for (Map.Entry<String, RuleAttribute> entry : ruleAttributes.entrySet()) {
+            if (coordinate.getCol() == entry.getValue().getIndex()) {
+                if (updateRelevantAttribute(entry.getKey())) {
+                    return true;
+                } else {
+                    isAttribute = true;
+                    break;
+                }
+            }
+        }
+        if (isAttribute) {
+            return false;
+        }
```

Comment:
I think could be simplified slightly, or am I missing something?
```java
public boolean update() {

        for (Map.Entry<String, RuleAttribute> entry : ruleAttributes.entrySet()) {
            if (coordinate.getCol() == entry.getValue().getIndex()) {
                if (updateRelevantAttribute(entry.getKey())) {
                    return true;
                } else {
                    return false;
                }
            }
        }
        
```

Reply:
You might not be, but I remember I had to play with this a bit and I did try a simple solution, but this was the one that worked. So afraid to touch it ATM.

Reply:
@jomarko Your proposal skips the remainder of the method. See [here](https://github.com/kiegroup/drools-wb/blob/e9b0fc416a8c0f0525115b19cc305d37940b7728/drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/main/java/org/drools/workbench/services/verifier/plugin/client/CellUpdateManagerBase.java#L96).

Reply:
@manstis sorry, do not agree this time. In the removed code, the for cycle set `isAttribute` to `true` and break the cycle. Immediately after the cycle we have `if` that return `false` from the method if the `isAttribute` was true. However am not going to fight here. Approving.